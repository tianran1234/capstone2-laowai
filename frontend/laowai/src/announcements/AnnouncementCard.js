import React from "react";
import { Link } from "react-router-dom";


/** Show limited information about an announcement
 *
 * Is rendered by AnnouncementList to show a "card" for each announcement.
 *
 * announcementList -> announcementCard
 */

function AnnouncementCard({ id, username, title, postedat }) {
  console.debug("announcementCard", id);

  return (
      <Link className="AnnouncementCard card" to={`/announcements/${id}`}>
        <div className="card-body">
          <h4 className="card-title">
            {title}
          </h4>
          <p>by {username} at {postedat}</p>
        </div>
      </Link>
  );
}


export default AnnouncementCard;
