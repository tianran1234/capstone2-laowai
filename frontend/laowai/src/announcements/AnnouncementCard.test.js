import React from "react";
import { render } from "@testing-library/react";
import AnnouncementCard from "./AnnouncementCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <AnnouncementCard
            id="1"
            username="testadmin"
            title="Panda Workshop"
            postedat="2023-08-15"
        />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});


