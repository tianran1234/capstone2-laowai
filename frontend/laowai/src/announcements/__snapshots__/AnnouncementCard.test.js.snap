// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="AnnouncementCard"
    href="/announcements/1"
  >
    <div
      class="card-body"
    >
      <h4
        class="card-title"
      >
        Panda Workshop
      </h4>
      <p>
        by testadmin at 2023-08-15
      </p>
    </div>
  </a>
</DocumentFragment>
`;
